framework/src/services.py

```diff
@@ -1649,6 +1650,19 @@ def stage_output_files(self, timeStamp, file_list, keep_old_files = True):
             self.exception('Error in stage_output_files()')
             raise
 
+#  Subworkflows can specify a OUTPUT directory. Copy files there.
+        if 'OUTPUT_DIR' in self.component_ref.config:
+            try:
+                ipsutil.copyFiles(workdir, all_files, self.component_ref.config['OUTPUT_DIR'],
+                                  outprefix, keep_old=keep_old_files)
+            except Exception, e:
+                self._send_monitor_event('IPS_STAGE_OUTPUTS',
+                                         'Files = ' + str(file_list) + \
+                                         ' Exception raised : ' + str(e),
+                                         ok='False')
+                self.exception('Error in stage_output_files()')
+                raise
+
         # Store plasma state files into $SIM_ROOT/history/plasma_state
```

Comment:
So this will copy all output files from all components in the sub flow in every step where stage_output_files() is  called to a single OUTPUT_DIR, is this really the right  design?. What if the sub-flow has several components and runs for tens of steps or more - do we want to copy all of these files and make them available to the super-flow?

Reply:
In the usage scenario I envision, the super workflow doesn't know or care what components staged output files. What I really need is just the current plasma state of the sub workflow.

Attached is some slides explaining how the workflows couple.

[Nested workflow use cases.pptx](https://github.com/HPC-SimTools/IPS-framework/files/1910819/Nested.workflow.use.cases.pptx)

Reply:
Check this call https://github.com/HPC-SimTools/IPS-framework/blob/d39232eb1e6c44b41f36e6f35ccc2011271e6cdd/framework/src/services.py#L1582 for a way to access subflow output from the super flow.